server/server.ts

```diff
@@ -107,6 +126,101 @@ app.get('/ping', (_req, res) => {
   res.json('pong');
 });
 
+app
+  .post('/discordRole', async (req, res) => {
+    const decoded = await validateUserToken(req.body?.uid, req.body?.token);
+    if (!decoded) {
+      return res.status(400).json({ error: 'invalid user token' });
+    }
+    const customer = await getCustomerByEmail(decoded.email as string);
+    const isSubscriber = Boolean(
+      customer?.subscriptions?.data?.find((sub) => sub?.status === 'active')
+    );
+    if (!isSubscriber) {
+      return res
+        .status(400)
+        .json({ error: 'You are not a WatchParty Plus subscriber.' });
+    }
+    if (!postgres || !discordBot) {
+      return res.status(500);
+    }
+    try {
+      if (req?.body.undo) {
+        const user = await discordBot.assignRole(
+          req.body?.username,
+          req.body?.discriminator,
+          true
+        );
+        if (user) {
+          await postgres?.query('DELETE FROM discord WHERE "customerId" = $1', [
```

Comment:
we could just store email + discord id + sub state? And we could update the sub state in the same code that already syncs subs to postgres

Reply:
But do we have to save anything if we never remove the sub roles like you suggested above?

Reply:
I assume users might want to delete/edit their discord linked account